pkg/provider/cloud.go

```diff
@@ -177,3 +182,16 @@ func (c *cloud) ProviderName() string {
 func (c *cloud) HasClusterID() bool {
 	return true
 }
+
+var getInfraKubeConfig = func(infraKubeConfigPath string) (string, error) {
```

Comment:
You don't have to use anonymous function. I understand you did this for tests but since you accept a path as a parameter for this function, you could do something similar like folks for configmap tests https://github.com/kubevirt/kubevirt/blob/main/pkg/config/config-map_test.go#L41, use a temp file and inject valid or invalid kubeconfig.